devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/dependency/LocalExtensionDependency.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.gradle.dependency;
+
+import java.util.List;
+
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+
+import io.quarkus.bootstrap.model.AppArtifactCoords;
+import io.quarkus.maven.dependency.ArtifactKey;
+
+public class LocalExtensionDependency extends ExtensionDependency {
+
+    private static final String DEFAULT_DEPLOYMENT_PATH_SUFFIX = "deployment";
+
+    private Project localProject;
+
+    public LocalExtensionDependency(Project localProject, ModuleVersionIdentifier extensionId,
+            AppArtifactCoords deploymentModule,
+            List<Dependency> conditionalDependencies, List<ArtifactKey> dependencyConditions) {
+        super(extensionId, deploymentModule, conditionalDependencies, dependencyConditions);
+        this.localProject = localProject;
+    }
+
+    public String findDeploymentModulePath() {
+
+        String deploymentModuleName = DEFAULT_DEPLOYMENT_PATH_SUFFIX;
+        if (localProject.getParent().findProject(deploymentModule.getArtifactId()) != null) {
```

Comment:
In which case could this return null?

Reply:
By convention the deployment module is called `deployment` but it can be whatever other name. Here I try to look for a project named with the artifact id. If it’s null I fallback to the conventional `deployment`. I could look for the deployment project first. Actually, are `runtime` / `deployment` module names mandatory?

Reply:
No, they shouldn't be.

Reply:
Is `deploymentModule.getArtifactId()` ever null or empty? if `localProject.getParent().findProject(deploymentModule.getArtifactId()` returns null, what does it mean?

Reply:
`deploymentModule.getArtifactId()` should never be null (however, it can be null if the `deployment-artifact` property is missing from `quarkus-extension.properties` files but that would mean the extension is corrupted).

if `localProject.getParent().findProject(deploymentModule.getArtifactId()` returns null, this means that in the current multi-module project, there is no project named using the artifact id (which may be always the case as we promote a module name `deployment` whereas the deployment artifact name is more something like `my-extension-deployment`).

Reply:
It sounds like if `deploymentModule.getArtifactId()` is null we should just throw an error.

>if localProject.getParent().findProject(deploymentModule.getArtifactId() returns null, this means that in the current multi-module project, there is no project named using the artifact id (which may be always the case as we promote a module name deployment whereas the deployment artifact name is more something like my-extension-deployment).

Could we have `deploymentModule.getModuleName()`? Or figure it out in some other way? It's ok to have a default for the module name but we also shouldn't be only guessing here. `deployment` as the default module name is reasonable, imo, btw.

Reply:
If we add that in the `META-INF/quarkus-extension.properties` file then yes. But it will requires the extension to use the `io.quarkus.extension` plugin. 
I though of adding a configuration entry in the `io.quarkus` plugin that will be limited to multi-module project with one extension. For example, it won't work with the following structure:

```
root
  application
  ext-a
    runtime
    deployment
  ext-b
    runtime
    deployment

```

Reply:
Ok, I think we can merge this for now. But we should look into making it more reliable.
E.g. we could check whether modules apply `io.quarkus.extension` plugin, deployment modules also contain extension metadata, e.g. `META-INF/quarkus-build-steps.list`. Couldn't we find which module produced a given artifact in the project (we know what the deployment artifact is, we just need to find its origin module)?